File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/location/GetLocationSettingsUseCase.java
package com.lattis.ellipse.domain.interactor.location;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.LocationSettingsResult;
import com.lattis.ellipse.domain.repository.LocationRepository;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.disposables.Disposable;
import io.reactivex.observers.DisposableObserver;

public class GetLocationSettingsUseCase extends UseCase<LocationSettingsResult> {

    private LocationRepository locationRepository;

    @Inject
    protected GetLocationSettingsUseCase(ThreadExecutor threadExecutor,
                                        PostExecutionThread postExecutionThread,
                                        LocationRepository locationRepository) {
        super(threadExecutor, postExecutionThread);
        this.locationRepository = locationRepository;
    }

    @Override
    protected Observable<LocationSettingsResult> buildUseCaseObservable() {
        return this.locationRepository.getLocationSettings();
    }

    @Override
    public Disposable execute(DisposableObserver<LocationSettingsResult> UseCaseSubscriber) {
        return this.buildUseCaseObservable()
                .subscribeWith(UseCaseSubscriber);
    }

}

File: velolabs_repos/operator-android/domain/src/main/java/io/lattis/domain/usecase/location/GetLocationSettingsUseCase.kt
package io.lattis.domain.usecase.location


import io.lattis.domain.executor.PostExecutionThread
import io.lattis.domain.executor.ThreadExecutor
import io.lattis.domain.models.LocationSettingsResult
import io.lattis.domain.repository.LocationRepository
import io.lattis.domain.usecase.base.UseCase
import javax.inject.Inject

import io.reactivex.Observable
import io.reactivex.disposables.Disposable
import io.reactivex.observers.DisposableObserver

class GetLocationSettingsUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            private val locationRepository: LocationRepository
) : UseCase<LocationSettingsResult>(threadExecutor, postExecutionThread) {

    override fun buildUseCaseObservable(): Observable<LocationSettingsResult> {
        return this.locationRepository.getLocationSettings()
    }

    override fun execute(UseCaseSubscriber: DisposableObserver<LocationSettingsResult>): Disposable {
        return this.buildUseCaseObservable()
                .subscribeWith(UseCaseSubscriber)
    }

}
